cre.cpp

```diff
@@ -2302,14 +2305,20 @@ static int drawCurrentPage(lua_State *L) {
                  * for why not TYPE_BBRGB32) */
 		LVColorDrawBuf drawBuf(w, h, bb->data, 16);
 		doc->text_view->Draw(drawBuf, false);
+		drawn_images_count = drawBuf.getDrawnImagesCount();
+		drawn_images_surface = drawBuf.getDrawnImagesSurface();
 	}
 	else {
 		/* Set DrawBuf to 8bpp */
 		LVGrayDrawBuf drawBuf(w, h, 8, bb->data);
 		doc->text_view->Draw(drawBuf, false);
+		drawn_images_count = drawBuf.getDrawnImagesCount();
+		drawn_images_surface = drawBuf.getDrawnImagesSurface();
 	}
 
-	return 0;
+	lua_pushinteger(L, drawn_images_count);
+	lua_pushnumber(L, (float)drawn_images_surface/(w*h));
+	return 2;
```

Comment:
This just returns to frontend code stats about images drawn on the page, so that @NiLuJe can get them with `Document:getDrawnImagesStatistics()` (that will return these 2 numbers for credocument, nil for others) and decide if he should use HW dithering after painting this page in ReaderView.

Reply:
Sure, but that requires return 2?

Reply:
Oh, yes, required :)
All these `return N` in cre.cpp indicate to the lua glue the number of things we put on its stack with lua_pushThing, so we get them in frontend (not tested, but I'm quite sure I won't get them if I return 0).

Reply:
Oh right, of course! You need to communicate the number of return values, I forgot all about that. :blush: